Sources/AblyChat/Errors.swift

```diff
@@ -16,37 +16,163 @@ public enum ErrorCode: Int {
     /// TODO this code is a guess, revisit in https://github.com/ably-labs/ably-chat-swift/issues/32
     case inconsistentRoomOptions = 1
 
+    case messagesAttachmentFailed = 102_001
+    case presenceAttachmentFailed = 102_002
+    case reactionsAttachmentFailed = 102_003
+    case occupancyAttachmentFailed = 102_004
+    case typingAttachmentFailed = 102_005
+
+    case messagesDetachmentFailed = 102_050
+    case presenceDetachmentFailed = 102_051
+    case reactionsDetachmentFailed = 102_052
+    case occupancyDetachmentFailed = 102_053
```

Comment:
**suggestion:** Consider using a more systematic approach for error code values

The current use of magic numbers for error codes, while grouped logically, could be improved. Consider using constants or a more descriptive naming convention to make the values' significance clearer and easier to maintain.